ql/math/statistics/riskstatistics.hpp

```diff
@@ -151,12 +151,12 @@ namespace QuantLib {
     template <class S>
     Real GenericRiskStatistics<S>::regret(Real target) const {
         // average over the range below the target
-        std::pair<Real,Size> result = this->expectationValue(
+        std::pair<Real, Size> result = this->expectationValue(
             [=](Real xi) -> Real {
                 Real d = (xi - target);
                 return d * d;
             },
-            [=](Real xi) -> Real { return xi < target; });
+            [=](Real xi) -> Real { return static_cast<QuantLib::Real>(xi < target); });
```

Comment:
@xcelerit-dev : this should have been `-> bool`, right?

Reply:
Indeed @lballabio, that was an oversight and should have been bool. We've carefully checked for similar things across the codebase: This is the only place where an inappropriate lambda return type was used. Would you like us to submit a new PR cleaning this issue (via our fork) or can you do this directly in this PR?

Reply:
I can fix it here, thanks for checking